projects/workflows-creator/src/lib/builder/builder.component.ts

```diff
@@ -429,8 +473,55 @@ export class BuilderComponent<E> implements OnInit, OnChanges {
    * It builds a new diagram, emits the new diagram, and then tells Angular to update the view
    */
   async updateDiagram() {
+    const nodes = [
+      ...this.eventGroups[0].children,
+      ...this.actionGroups[0].children,
+      ...this.elseActionGroups[0].children,
+    ];
+    let isValid =
+      !!this.eventGroups[0].children.length &&
+      (!!this.actionGroups[0].children.length ||
+        !!this.elseActionGroups[0].children.length);
+    if (isValid) {
+      for (const node of nodes) {
```

Comment:
for validation, we can use a separate service that gets the current state of diagram and just returns true or false and use that here, and this should be based on the default inputs provided. Anyone using the library should be able to replace it to write there own logic according to there own custom events/actions/inputs